src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -294,19 +294,23 @@ private string ModifyWorkloadSet(string advertisingPackagePath, InstallAction re
 
             InstallAction plannedAction = PlanPackage(msi, state, requestedAction, installedVersion);
 
-            if (plannedAction != InstallAction.None)
+            if (plannedAction == InstallAction.Uninstall)
+            {
+                Elevate();
+
+                // Update the reference count against the MSI.
+                UpdateDependent(InstallRequestType.RemoveDependent, msi.Manifest.ProviderKeyName, _dependent);
+
+                ExecutePackage(msi, plannedAction, msiPackageId);
```

Comment:
Looks like per our discussion with Jacques, this ordering should fix the rollback failures.